shared/common-adapters/drop-file-box.desktop.js

```diff
@@ -0,0 +1,96 @@
+// @flow
+import * as React from 'react'
+import fs from 'fs'
+import logger from '../logger'
+import Icon from './icon'
+import {Box, Box2} from './box'
+import {globalStyles, globalColors} from '../styles'
+
+type State = {|
+  showDropOverlay: boolean,
+|}
+
+const DropOverlay = ({onDragLeave, onDrop}) => (
+  <Box style={dropOverlayStyle} onDragLeave={onDragLeave} onDrop={onDrop}>
+    <Icon type="icon-dropping-file-48" />
+  </Box>
+)
+
+class DropFileBox extends React.Component<any, State> {
+  state = {showDropOverlay: false}
+
+  _onDrop = e => {
+    if (!this._validDrag(e)) {
+      return
+    }
+    const fileList = e.dataTransfer.files
+    const paths = fileList.length ? Array.prototype.map.call(fileList, f => f.path) : []
+    if (paths.length) {
+      for (let path of paths) {
+        // Check if any file is a directory and bail out if not
+        try {
+          // We do this synchronously
+          // in testing, this is instantaneous
+          // even when dragging many files
+          const stat = fs.lstatSync(path)
```

Comment:
I wonder if we need to consider the possibility that this could be from a KBFS mount, in which case it'd be slow. I guess in that case we could probably do a link for chat, and copy for Files. Not sure though!